sample/rules.go

```diff
@@ -70,6 +72,16 @@ func (s *RulesBasedSampler) GetSampleRate(trace *types.Trace) (rate uint, keep b
 			for _, span := range trace.GetSpans() {
 				var match bool
 				value, exists := span.Data[condition.Field]
+				if !exists {
```

Comment:
I'm concerned this would impact users that does not want to pay the cost of attempting to JSON parse span data when trying to resolve rule conditions.

I think we should look into adding an opt-in configuration option to enable this behaviour. Default would be disabled.

Reply:
That makes sense. I can add an optional parameter in the rules config file to enable it (similar to the AddSampleRateKeyToTrace and UseTraceLength fields), unless you think there is a better place for it.